kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -336,11 +337,8 @@ protected void deactivate(ComponentContext componentContext) {
     // Don't perform internal calls to this method
     @Override
     public ComponentConfiguration getComponentConfiguration(String pid) throws KuraException {
-        ComponentConfiguration tempConfig = getComponentConfigurationInternal(pid);
-        if (tempConfig != null && tempConfig.getConfigurationProperties() != null) {
-            decryptPasswords(tempConfig);
-        }
-        return tempConfig;
+        final ComponentConfiguration tempConfig = getComponentConfigurationInternal(pid);
+        return decryptPasswords(tempConfig, this.m_cryptoService);
```

Comment:
There is no need as methods called with the parameter can handle `null` as an argument. See `decryptPasswords` -> `decrypt`